asv_bench/benchmarks/benchmarks.py

```diff
@@ -78,33 +80,58 @@ def time_groupby_agg_mean(self, data_size, count_columns):
         return self.df.groupby(by=self.groupby_columns).apply(lambda df: df.mean())
 
 
-class TimeGroupByDefaultAggregations:
+class TimeGroupByDefaultAggregations(BaseTimeGroupBy):
     param_names = ["data_size"]
     params = [
         UNARY_OP_DATA_SIZE,
     ]
 
-    def setup(self, data_size):
-        self.df = generate_dataframe(
-            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
-        )
-        self.groupby_column = self.df.columns[0]
-
     @trigger_execution
     def time_groupby_count(self, data_size):
-        return self.df.groupby(by=self.groupby_column).count()
+        return self.df.groupby(by=self.groupby_columns).count()
 
     @trigger_execution
     def time_groupby_size(self, data_size):
-        return self.df.groupby(by=self.groupby_column).size()
+        return self.df.groupby(by=self.groupby_columns).size()
 
     @trigger_execution
     def time_groupby_sum(self, data_size):
-        return self.df.groupby(by=self.groupby_column).sum()
+        return self.df.groupby(by=self.groupby_columns).sum()
 
     @trigger_execution
     def time_groupby_mean(self, data_size):
-        return self.df.groupby(by=self.groupby_column).mean()
+        return self.df.groupby(by=self.groupby_columns).mean()
+
+
+class TimeGroupByDictionaryAggregation(BaseTimeGroupBy):
+    param_names = ["data_size"]
+    params = [
+        UNARY_OP_DATA_SIZE,
+    ]
+    reduction_operations = ["sum", "count", "prod"]
+    agg_operations = ["quantile", "std", "median"]
+
+    def setup(self, data_size):
+        super().setup(data_size)
+        self.cols_to_agg = self.df.columns[1:4]
+
+    @trigger_execution
+    def time_groupby_dictionary_reduction(self, data_size):
+        return self.df.groupby(by=self.groupby_columns).agg(
+            {
+                c: self.reduction_operations[i % len(self.reduction_operations)]
+                for i, c in enumerate(self.cols_to_agg)
+            }
```

Comment:
Can we move dictionary creation out of time_* function?

Reply:
done, now I create that dictionary in the `setup` function going by `operation_type` parameter